src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCaptureCollection.cs

```diff
@@ -47,6 +50,25 @@ public int Count
         }
 
         /// <summary>
+        /// Copies all the elements of the collection to the given array
+        /// beginning at the given index.
+        /// </summary>
+        public void CopyTo(Capture[] array, int arrayIndex)
+        {
+            if (array == null)
+                throw new ArgumentNullException("array");
+            if (arrayIndex < 0 || arrayIndex > array.Length)
```

Comment:
I originally had it like that, but then the behavior differs from `List<T>` and `Dictionary<TKey, TValue>`, which throws `ArgumentException` in that case instead of `ArgumentOutOfRangeException`.

See https://github.com/dotnet/coreclr/blob/ef1e2ab328087c61a6878c1e84f4fc5d710aebce/src/mscorlib/src/System/Collections/Generic/Dictionary.cs#L818